src/main/java/org/ohdsi/webapi/security/PermissionService.java

```diff
@@ -285,9 +285,9 @@ public boolean hasReadAccess(CommonEntity entity) {
             try {
                 String login = this.permissionManager.getSubjectName();
                 UserSimpleAuthorizationInfo authorizationInfo = this.permissionManager.getAuthorizationInfo(login);
-                if (Objects.equals(authorizationInfo.getUserId(), entity.getCreatedBy().getId())){
-		    hasAccess = true; // the role is the one that created the artifact
-		} else {
+                // if (Objects.equals(authorizationInfo.getUserId(), entity.getCreatedBy().getId())){
+                //     hasAccess = true; // the role is the one that created the artifact - Is this for backwards compatibiliy mode? Should we write a migration instead?
+		        // } else {
```

Comment:
TODO - make this configurable: this gives the user access to all artifacts he creates. We only want the user to see the artifacts he has been granted access through a role... and disregard createdBy

Reply:
use `if (this.authorizationMode.equals("teamproject"))`

Reply:
done